hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/testutils/SparkClientFunctionalTestHarness.java

```diff
@@ -425,7 +426,7 @@ protected HoodieWriteConfig.Builder getConfigBuilder(Boolean autoCommit, Boolean
 
   protected Dataset<Row> toDataset(List<HoodieRecord> records, Schema schema) {
     List<GenericRecord> avroRecords = records.stream()
-        .map(r -> (GenericRecord) r.getData())
+        .map(r -> (GenericRecord) ((SerializableIndexedRecord)r.getData()).getData())
```

Comment:
`SerializableIndexedRecord` also implements `GenericRecord`, why do we need these extra steps?